### PR TITLE
Add support for upto and offset in JSON files

### PR DESCRIPTION
**Pull request by mmicko**
Depends on : https://github.com/YosysHQ/yosys/pull/1124

### CODE REVIEW

File: json/jsonparse.cc [hunk 1]
```diff
@@ -406,8 +406,11 @@ void json_import_ports(Context *ctx, const string &modname, const std::vector<Id
             wire_node = wire_group_node->data_array[index];
             //
             // Pick a name for this port
+            int ndx = index + start_offset;
+            if (!upto)
```
Comment (daveshah1) [inline json/jsonparse.cc]: This looks like the wrong way round to me - when upto isn't set, we surely want to keep the old behaviour?

Comment (mmicko) [reply json/jsonparse.cc]: Agree, but in that case yosys should always set "upto" to 1 or 0, right now we can not distinguish between 0 and old behavior. cc: @cliffordwolf

Comment (daveshah1) [reply json/jsonparse.cc]: We should certainly not be silently breaking JSON files from older Yosys

File: json/jsonparse.cc [hunk 2]
```diff
@@ -732,6 +747,18 @@ void json_import(Context *ctx, string modname, JsonNode *node)
 
             here = cell_parent->data_dict.at(cell_parent->data_dict_keys[nnid]);
             std::string basename = cell_parent->data_dict_keys[nnid];
+            bool upto = false;
+            int start_offset = 0;
+            if (here->data_dict.count("upto") != 0) {
+				JsonNode *val = here->data_dict.at("upto");
+				if (val->type == 'N')
+					upto = val->data_number != 0;
+			}
```
Comment (cliffordwolf) [inline json/jsonparse.cc]: This indenting doesn't look right..  ;)

Comment (mmicko) [reply json/jsonparse.cc]: Fixed now. Thanks for pointing out to this.

**Pull request merged**